modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java

```diff
@@ -150,8 +151,13 @@ public static void  invokeResource(CarbonMessage cMsg, CarbonCallback callback,
         Context context = new Context(programFile);
         context.setServiceInfo(serviceInfo);
         context.setCarbonMessage(resourceMessage);
-        context.setProperty("SRC_HANDLER", resourceMessage.getProperty("SRC_HANDLER"));
         context.setBalCallback(new DefaultBalCallback(resourceCallback));
+
+        SourceHandler srcHandler = null;
+        if (resourceMessage.getProperty(Constants.SRC_HANDLER) != null) {
+            srcHandler = (SourceHandler) resourceMessage.getProperty(Constants.SRC_HANDLER);
+            context.setProperty(Constants.SRC_HANDLER, srcHandler);
+        }
```

Comment:
Now that we set the SH here, I suppose we don't have to set it below

https://github.com/ballerinalang/ballerina/blob/master/modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/actions/http/AbstractHTTPAction.java#L144